cmd/agent/api/security.go

```diff
@@ -67,26 +58,23 @@ func buildSelfSignedKeyPair() ([]byte, []byte) {
 	return rootCertPEM, rootKeyPEM
 }
 
-func initializeTLS() error {
-	cert, key := buildSelfSignedKeyPair()
+func initializeTLS(additionalHostIdentities ...string) (*tls.Certificate, *x509.CertPool, error) {
+	log.Info("Initializing TLS certificates")
```

Comment:
nit: Related to the original comment around logging, it would be good to identify what TLS certificates we're initializing, probably by the caller ID/name so that we know the requestor of the cert. I know we print out the hosts with the name in securiity.go but that log line could be far away from this one in the output.

Reply:
I'll add that in a separate PR where we can discuss what kind of information we want around TLS initialization